src/libstd/net/tcp.rs

```diff
@@ -780,17 +809,56 @@ impl TcpListener {
 
     /// Moves this TCP stream into or out of nonblocking mode.
     ///
-    /// On Unix this corresponds to calling fcntl, and on Windows this
-    /// corresponds to calling ioctlsocket.
+    /// This will result in `read`, `write`, `recv` and `send` operations
+    /// becoming nonblocking, i.e. immediately returning from their calls.
+    /// If the IO operation is successful, `Ok` is returned and no further
+    /// action is required. If the IO operation could not be completed and needs
+    /// to be retried, an error with kind [`io::ErrorKind::WouldBlock`] is
+    /// returned.
+    ///
+    /// On Unix platforms, calling this method corresponds to calling `fcntl`
+    /// `FIONBIO`. On Windows calling this method corresponds to calling
+    /// `ioctlsocket` `FIONBIO`.
     ///
     /// # Examples
     ///
+    /// Bind a TCP listener to an address, listen for connections, and write
+    /// bytes in nonblocking mode:
+    ///
     /// ```no_run
+    /// use std::io::{self, Write};
     /// use std::net::TcpListener;
     ///
-    /// let listener = TcpListener::bind("127.0.0.1:80").unwrap();
+    /// let listener = TcpListener::bind("127.0.0.1:7878").unwrap();
     /// listener.set_nonblocking(true).expect("Cannot set non-blocking");
+    ///
+    /// # fn wait_for_fd() { unimplemented!() }
+    /// for stream in listener.incoming() {
```

Comment:
Would it be simpler to just open a socket and write to it rather than also including a listener?

Reply:
this doc comment is for`TcpListener::set_nonblocking`, so it made sense for me to include the listener. can you expand on what you mean?

Reply:
unless you're saying i shouldn't write to the stream. i could just read from it and print out the result. i'd also avoid the `write_all` thing from your other comment

Reply:
Oh derp, sorry. I was thinking this was on `write` for some reason...